Dynamically determine the correct Images account to use

jsf9k opened the pull request:
## 🗣 Description

In this pull request I dynamically determine the correct Images account to use when searching for AMIs.  I also switch to using the new COOL DNS account for public DNS, and I update the code to adapt to changes made in cisagov/openvpn-server-tf-module#19.

## 💭 Motivation and Context

This change is required to deploy a COOL staging environment.

## 🧪 Testing

I used these changes, together with those in cisagov/openvpn-server-tf-module#19, to deploy a COOL staging environment.

## 🚥 Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

jsf9k committed:
Split out the caller identity into its own file

It's a little difficult to find when it's hidden at the top of the
openvpn.tf file.

jsf9k committed:
Add a resource so we can look up information about organization accounts

jsf9k committed:
Use the new COOL DNS account for public DNS

jsf9k committed:
Dynamically determine the correct Images account ID

dav3r reviewed (approved):
This all feels so... familiar. ♊️

felddy reviewed (approved):
A joy to have in class.  🌞

jsf9k committed:
Update code to agree with recent changes to cisagov/openvpn-server-tf-module

Specifically, cisagov/openvpn-server-tf-module#19.  This involved:
* Adding a remote state from which to pull the public hosted zone ID
* Removing the "domain"" input variable from the
cisagov/openvpn-server-tf-module configuration
* Adding the "public_zone_id" variable to the
cisagov/openvpn-server-tf-module configuration

jsf9k committed:
Remove now-unused public_zone_name input variable

felddy reviewed (approved):
Looking good.  Thanks for letting me review it.
👍

dav3r reviewed (approved):
And the beat goes on! 🥁
